socket/control.go

```diff
@@ -0,0 +1,84 @@
+package socket
```

Comment:
This makes perfect sense, removing impl from naming. This was one of those 2am smart ideas as I had it as `control` originally. 😁